litex/soc/cores/cpu/rocket/core.py

```diff
@@ -44,89 +44,50 @@
-    "linux"    : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "linux4"   : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "linuxd"   : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "linux2d"  : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "linuxq"   : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "linux2q"  : "-march=rv64i2p0_mac   -mabi=lp64 ",
-    "full"     : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "fulld"    : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "full4d"   : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "fullq"    : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "full4q"   : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "fullo"    : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "full4o"   : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-    "full8o"   : "-march=rv64i2p0_mafdc -mabi=lp64 ",
-}
-
-# CPU Params ----------------------------------------------------------------------------------
-
-CPU_PARAMS = {
-    # Variant  : (mem_dw, mmio_dw, num_cores)
-    "standard" : (    64,      64,         1),
-    "linux"    : (    64,      64,         1),
-    "linux4"   : (    64,      64,         4),
-    "linuxd"   : (   128,      64,         1),
-    "linux2d"  : (   128,      64,         2),
-    "linuxq"   : (   256,      64,         1),
-    "linux2q"  : (   256,      64,         2),
-    "full"     : (    64,      64,         1),
-    "fulld"    : (   128,      64,         1),
-    "full4d"   : (   128,      64,         4),
-    "fullq"    : (   256,      64,         1),
-    "full4q"   : (   256,      64,         4),
-    "fullo"    : (   512,      64,         1),
-    "full4o"   : (   512,      64,         4),
-    "full8o"   : (   512,      64,         8),
-}
-
 # Rocket  ------------------------------------------------------------------------------------------
 
 class Rocket(CPU):
     category             = "softcore"
     family               = "riscv"
     name                 = "rocket"
     human_name           = "RocketRV64[imac]"
-    variants             = CPU_VARIANTS
+    variants             = ["small", "medium", "linux", "full"]
     data_width           = 64
     endianness           = "little"
     gcc_triple           = CPU_GCC_TRIPLE_RISCV64
     linker_output_format = "elf64-littleriscv"
     nop                  = "nop"
     io_regions           = {0x1200_0000: 0x7000_0000} # Origin, Length.
 
+    # Default parameters.
+    core_count = 1
+    mem_width  = 1
+
+    # Command line configuration arguments.
+    @staticmethod
+    def args_fill(parser):
+        cpu_group = parser.add_argument_group(title="CPU options")
+        cpu_group.add_argument("--core-count", default=1, help="Number of cores (1, 2, 4, 8).", type=int)
+        cpu_group.add_argument("--mem-width",  default=1, help="Width of memory port (1, 2, 4, 8).", type=int)
+
+    @staticmethod
+    def args_read(args):
+        if int(args.core_count) in [1, 2, 4, 8]:
+            Rocket.core_count = int(args.core_count)
+        else:
+            Rocket.core_count = 1 # FIXME: warn user!!!
```

Comment:
I haven't seen an example of another CPU family logging warnings or errors, but it'd be nice to send out a LiteX-y warning here if the user requests any other core count (or memory port width, see below) than 1, 2, 4, or 8, for which we have pre-built verilog... :)